pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+            and self._precision_plugin_flag
+            and not isinstance(self._precision_plugin_flag, (TPUPrecisionPlugin, TPUBf16PrecisionPlugin))
+        ):
+            raise ValueError(
+                f"The `TPUAccelerator` can only be used with a `TPUPrecisionPlugin`,"
+                f" found: {self._precision_plugin_flag}."
             )
-        elif self.use_dp:
-            plugin = DataParallelStrategy(parallel_devices=self.parallel_devices)
-        elif self.use_horovod:
-            plugin = HorovodStrategy(parallel_devices=self.parallel_devices)
-        elif self.use_tpu and isinstance(self.tpu_cores, list):
-            plugin = SingleTPUStrategy(self.tpu_id)
-        elif self.use_ipu:
-            plugin = IPUStrategy(parallel_devices=self.parallel_devices)
-        else:
-            single_gpu_ordinal = device_parser.determine_root_gpu_device(self.parallel_device_ids)
-            plugin = SingleDeviceStrategy(device=single_gpu_ordinal if self.use_gpu else "cpu")
-        return plugin
-
-    def resolve_strategy(self, training_type: Strategy) -> Strategy:
-        # necessary for when the user has passed in a plugin
-        if hasattr(training_type, "parallel_devices") and getattr(training_type, "parallel_devices") is None:
-            training_type.parallel_devices = self.parallel_devices
-
-        if hasattr(training_type, "cluster_environment") and getattr(training_type, "cluster_environment") is None:
-            # transfer ownership of the cluster environment to the training type
-            training_type.cluster_environment = self.cluster_environment
-            self._cluster_environment = proxy(self.cluster_environment)
-
-        if hasattr(training_type, "num_nodes"):
-            # set num_nodes for training_type from trainer setting
-            training_type.num_nodes = self.num_nodes
-
-        if hasattr(training_type, "sync_batchnorm"):
-            # set sync_batchnorm for training_type from trainer setting
-            training_type.sync_batchnorm = self.sync_batchnorm
-
-        return training_type
-
-    def select_accelerator(self) -> Accelerator:
-        if isinstance(self.distributed_backend, Accelerator):
-            # custom accelerator from user
-            if self._precision_plugin is not None or self._strategy is not None:
-                # plugins also specified by user
-                rank_zero_warn(
-                    "Specified `Precision` and `TrainingType` plugins will be ignored,"
-                    " since an `Accelerator` instance was provided."
+        if (
+            self._precision_flag == 16
+            and isinstance(self.accelerator, CPUAccelerator)
+            and self._amp_type_flag == AMPType.APEX
+        ):
+            raise MisconfigurationException(
+                "You passed `Trainer(accelerator='cpu', precision=16, amp_type='apex')`"
+                " but apex AMP not supported on CPU."
```

Comment:
Worth to mention this works with bfloat16 and native.

Reply:
added to #11449